portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -90,20 +92,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Take the input from the POST request 
-    String text = request.getParameter("text-input");
-    String author = request.getParameter("author");
-    // The added date for the comment will be the current date.
-    Date date = new Date();
-
-    Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("text", text);
-    commentEntity.setProperty("author", author);
-    commentEntity.setProperty("date", date);
-
-    // Store comment data in the database
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentEntity);
+    // Check that the user is authenticated before adding the comment
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (userService.isUserLoggedIn()) {
+      // Take the input from the POST request 
+      String text = request.getParameter("text-input");
+      String email = userService.getCurrentUser().getEmail();
+      // The added date for the comment will be the current date.
+      Date date = new Date();
+
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("text", text);
+      commentEntity.setProperty("author", email);
```

Comment:
Email address is considered as PII. It can be stored in the comment. But it should be exposed through APIs

https://www.google.com/search?q=pii&oq=pii&aqs=chrome..69i57j0l7.2776j0j1&sourceid=chrome&ie=UTF-8